ui/imports/shared/status/StatusChatInput.qml

```diff
@@ -717,9 +717,8 @@ Rectangle {
             insertInTextInput(d.copyTextStart, d.copiedTextFormatted)
             messageInputField.cursorPosition = d.copyTextStart + messageInputField.length - prevLength
             d.internalPaste = false
-        } else if (event.matches(StandardKey.Paste)) {
-            insertInTextInput(d.copyTextStart, QClipboardProxy.text)
-            messageInputField.cursorPosition = d.copyTextStart + QClipboardProxy.text.length
+        } else if (event.matches(StandardKey.Paste) && QClipboardProxy.hasText) {
+            messageInputField.insert(d.copyTextStart, "<div style='white-space: pre-wrap'>" + globalUtils.escapeHtml(QClipboardProxy.text) + "</div>") // preserve formatting
```

Comment:
Please consider using Utils.qml to access globalUtils. Also, there is an escapeHtml in Utils.qml that does the same thing in a different way

Reply:
> even though I noticed `globalUtils` is used directly in other places too in this component
True. If we're going to try replacing the globalUtils context property with singleton(and hopefully we will get there) it's going to be more work for us with each direct use of the context property. IMO it would be nice to change it everywhere in StatusChatInput, but it's up to you as it's not in the scope of this fix.

Totally agree that QString::toHtmlEscaped is better than the JS regex. I think it might be worth replacing the regex with the QString implementation. Adding @stefandunca in the loop in case we're missing something here.

Reply:
Ok, done :)